src/drivers/aerofc_adc/aerofc_adc.cpp

```diff
@@ -145,23 +168,27 @@ int AEROFC_ADC::init()
 
 int AEROFC_ADC::probe()
 {
-	uint8_t buffer[1];
+	uint8_t buffer[2];
 	int ret;
 
 	_retries = 3;
 
-	set_address(SLAVE_ADDR | ENABLE_ADC_REG);
-	buffer[0] = 1;
-	ret = transfer(buffer, sizeof(buffer), NULL, 0);
+	/* Enable ADC */
+	buffer[0] = ADC_ENABLE_REG;
+	buffer[1] = 0x01;
+	ret = transfer(buffer, 2, NULL, 0);
 
 	if (ret != PX4_OK) {
 		goto error;
 	}
 
-	buffer[0] = 0;
-	ret = transfer(0, 0, buffer, sizeof(buffer));
+	usleep(10000);
+
+	/* Read ADC value */
+	buffer[0] = ADC_CHANNEL_REG;
+	ret = transfer(buffer, 1, buffer, 2);
 
-	if (ret != PX4_OK || buffer[0] != 1) {
```

Comment:
It's just ensuring an initial transfer works. We don't know what value it will have. We just read a sample and check the return code, meaning we got any data from the sensor for that channel.